### PR TITLE
Add credo to CI tests

### LINKED ISSUE #100: Add credo for linting/style checking
Credo can be found at:

https://github.com/rrrene/credo

Whoever takes this should ensure all credo warnings/errors are cleaned up

### PR DESCRIPTION
**Pull request by joshuawscott**
Fixes #100

### CODE REVIEW

**Comment by bjhaid** [comment]
does ebert not provide the same functionality as credo?

**Comment by joshuawscott** [comment]
It does, but this gives the extra layer of ensuring that credo passes, or it will fail the test suite. It's currently passing, which was our roadblock in the past to enabling it, if I recall correctly.

File: scripts/ci_tests.sh [hunk 1]
```diff
@@ -6,8 +6,15 @@
 #
 # This script could be used for local testing as long as COVERALLS is not set.
 
+set -e
```
Comment (dantswain) [inline scripts/ci_tests.sh]: Instead of this, do we want to run both and then have it report which things failed?

Comment (bjhaid) [reply scripts/ci_tests.sh]: `set -ex` probably will be more obvious on which failed

File: scripts/ci_tests.sh [hunk 2]
```diff
@@ -16,16 +23,7 @@ else
...
-#    (we should strive to remove this but it is necessary for now)
-if [ $? -eq 0 ]
-then
-  echo "First tests passed, skipping repeat"
-else
-  echo "Repeating tests"
-  mix "$TEST_COMMAND" $INCLUDED_TESTS
-fi
+# Retry if it doesn't work the first time
+mix "$TEST_COMMAND" $INCLUDED_TESTS || mix "$TEST_COMMAND" $INCLUDED_TESTS
```
[the remark this answers is not in the document]
Comment (joshuawscott) [reply scripts/ci_tests.sh]: This is incorrect. Quoting this changes the meaning of the arguments.

File: scripts/ci_tests.sh (hunk 1 above)
Comment (joshuawscott) [reply scripts/ci_tests.sh]: done

**Pull request merged**